brainscore/benchmarks/imagenet_c.py

```diff
@@ -37,77 +37,77 @@
    url           = "https://arxiv.org/abs/1903.12261"
 }"""
 
+def Imagenet_C_Noise(sampling_factor=10):
+    return Imagenet_C_Category('noise', sampling_factor=sampling_factor)
 
-def Imagenet_C_Noise():
-    return Imagenet_C_Category('noise')
+def Imagenet_C_Blur(sampling_factor=10):
+    return Imagenet_C_Category('blur', sampling_factor=sampling_factor)
 
+def Imagenet_C_Weather(sampling_factor=10):
+    return Imagenet_C_Category('weather', sampling_factor=sampling_factor)
 
-def Imagenet_C_Blur():
-    return Imagenet_C_Category('blur')
-
-
-def Imagenet_C_Weather():
-    return Imagenet_C_Category('weather')
-
-
-def Imagenet_C_Digital():
-    return Imagenet_C_Category('digital')
+def Imagenet_C_Digital(sampling_factor=10):
+    return Imagenet_C_Category('digital', sampling_factor=sampling_factor)
 
 
 class Imagenet_C_Category(BenchmarkBase):
     """
-    Runs all ImageNet C benchmarks within a perturbation category, ie: 
+    Runs all ImageNet C benchmarks within a noise category, ie: 
     gaussian noise [1-5]
     shot noise [1-5]
     impulse noise [1-5]
     """
+    noise_category_map = {
+        'noise' : ['gaussian_noise', 'shot_noise', 'impulse_noise'],
+        'blur' : ['glass_blur', 'motion_blur', 'zoom_blur', 'defocus_blur'],
+        'weather' : ['snow', 'frost', 'fog', 'brightness'],
+        'digital' : ['pixelate', 'contrast', 'elastic_transform', 'jpeg_compression']
+    }
+
+    def __init__(self, noise_category, sampling_factor=10):
+        self.noise_category = noise_category
+        self.stimulus_set_name = f'dietterich.Hendrycks2019.{noise_category}'
+        
+        # take every nth image, n=sampling_factor.
+        stimulus_set = brainscore.get_stimulus_set(self.stimulus_set_name)[::sampling_factor]
+        self.stimulus_set = stimulus_set
+        self.noise_types = self.noise_category_map[noise_category]
 
-    def __init__(self, category):
-        category_groups = {
-            'noise': ['gaussian_noise', 'shot_noise', 'impulse_noise'],
-            'blur': ['glass_blur', 'motion_blur', 'zoom_blur', 'defocus_blur'],
-            'weather': ['snow', 'frost', 'fog', 'brightness'],
-            'digital': ['pixelate', 'contrast', 'elastic_transform', 'jpeg_compression']
-        }
-        self._category = category
-        self._groups = category_groups[category]
         ceiling = Score([1, np.nan], coords={'aggregation': ['center', 'error']}, dims=['aggregation'])
-        super(Imagenet_C_Category, self).__init__(identifier=f'dietterich.Hendrycks2019-{category}-top1', version=1,
+        super(Imagenet_C_Category, self).__init__(identifier='dietterich.Hendrycks2019-top1', version=1,
```

Comment:
I believe this should remain at `identifier=f'dietterich.Hendrycks2019-{category}-top1'`